CI: add Kernel compile tests

aparcar opened the pull request:
Add Github Actions yaml script to build test kernel PR changes for
each target.

Signed-off-by: Paul Spooren <mail@aparcar.org>
[ add commit description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

aparcar committed:
CI: add Kernel compile tests

Add Github Actions yaml script to build test kernel PR changes for
each target.

Signed-off-by: Paul Spooren <mail@aparcar.org>
[ add commit description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@aparcar was thinking... why not create a container instead of running debian and installing the package every time? It should speedup the execution

aparcar commented:
Problem is the broken external toolchain. That should be fixed first.

aparcar committed:
CI: run inside the buildbot docker container

Run github actions insider buildbot docker container.

Signed-off-by: Paul Spooren <mail@aparcar.org>
[ run container under buildbot user ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@aparcar I just notice that on other action we use `gosu`... here I use another solution as i wasn't aware of that... Don't know what is better.... readability or the contraption to have steps that run on the user directly...

Anyway I tested this on my local repo and in theory it should complete correctly.

I pushed the required changes for external toolchain to work.

Ansuel commented:
Mh 30 min each to test a kernel in not that bad... guess this needs some tweaks and it's ready...

would be very useful for kernel bump

aparcar commented on .github/workflows/kernel.yml:
```diff
@@ -95,45 +67,51 @@ jobs:
 
       - name: Update & Install feeds
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
```
smooth!

Ansuel replied:
Yep no idea if we should use gosudo or this to change the entire shell...imho this is more safe

aparcar commented:
I think this is great! Could you please cherry pick this into master (except the last commit)?

Ansuel commented:
Yes sure I will do this in the night as I have to rebase this with the changes for the external toolchain.

Ansuel commented:
Merged in master with:
8a77adb048 CI: run inside the buildbot docker container
c17c931a90 CI: add Kernel compile tests
